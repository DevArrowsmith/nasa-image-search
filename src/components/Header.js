import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import HeaderImage from './HeaderImage';
import SearchInput from './SearchInput';

const HeaderStyles = styled.div`
  height: 60px;
  width: 100vw;
  margin: 0;
  padding: 0 10px;
  background: #404138;
  position: sticky;
  top: 0px;
  display: flex;
  flex-flow: row nowrap;
  justify-content: start;
  align-items: center;
`;

const HeaderText = styled.p`
  margin: 0 5px 0 5px;
  width: 230px;
  font-size: 2.2em;
  color: red;
`
const Header = ({ setSearchResults, setResultsLoading }) => {

  return (
    <HeaderStyles>
      <HeaderImage 
        src="./images/nasa.png" 
        alt="NASA logo"
      />
      <HeaderText>Image Search</HeaderText>
      <SearchInput
        setSearchResults={setSearchResults}
        setResultsLoading={setResultsLoading}
       />
    </HeaderStyles>
  );

};

Header.propTypes = {
  setSearchResults: PropTypes.func,
  setResultsLoading: PropTypes.func
};

export default Header;
