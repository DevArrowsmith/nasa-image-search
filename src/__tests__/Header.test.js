import React from 'react';
import { render} from '@testing-library/react';
import Header from '../components/Header';

test('component Header renders', () => {
  const {asFragment } = render(<Header/>);
  const component = asFragment();
  expect(component).toMatchSnapshot();
});