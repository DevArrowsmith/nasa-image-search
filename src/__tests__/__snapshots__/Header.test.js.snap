// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Header renders 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw eJAjmE"
  >
    <img
      alt="NASA logo"
      class="sc-AxjAm ivsrOq"
      data-testid="nasa-logo-1"
      src="./images/nasa.png"
    />
    <input
      class="sc-AxirZ iviowO"
      placeholder="Search"
      type="text"
    />
  </div>
</DocumentFragment>
`;
